sql/src/main/java/io/crate/analyze/PrivilegesAnalyzer.java

```diff
@@ -97,17 +144,35 @@ private static void ensureUserManagementEnabled(@Nullable User user) {
     }
 
     private static Set<Privilege> privilegeTypesToPrivileges(Collection<Privilege.Type> privilegeTypes,
-                                                      User grantor,
-                                                      State state) {
+                                                             User grantor,
+                                                             State state,
+                                                             List<QualifiedName> tableOrSchemaNames,
+                                                             Privilege.Clazz clazz) {
         Set<Privilege> privileges = new HashSet<>(privilegeTypes.size());
-        for (Privilege.Type privilegeType : privilegeTypes) {
-            Privilege privilege = new Privilege(state,
-                privilegeType,
-                Privilege.Clazz.CLUSTER,
-                null,
-                grantor.name()
-            );
-            privileges.add(privilege);
+        if (Privilege.Clazz.CLUSTER.equals(clazz)) {
+            for (Privilege.Type privilegeType : privilegeTypes) {
+                Privilege privilege = new Privilege(state,
+                    privilegeType,
+                    Privilege.Clazz.CLUSTER,
+                    null,
+                    grantor.name()
+                );
+
+                privileges.add(privilege);
+            }
+        } else {
+            for (Privilege.Type privilegeType : privilegeTypes) {
+                for (QualifiedName name : tableOrSchemaNames) {
```

Comment:
i'm not sure how this is different .. I already check for the class and create the privilege when ident is null